python/hail/matrixtable.py

```diff
@@ -1687,13 +1691,20 @@ def group_rows_by(self, *exprs, **named_exprs):
             if any(not isinstance(a, Reference) and not isinstance(a, Select) for a in ast):
                 raise ExpressionException("method 'group_rows_by' expects keyword arguments for complex expressions")
             key = ast[0].name if isinstance(ast[0], Reference) else ast[0].selection
-            groups.append((key, e))
-        for k, e in named_exprs.items():
-            e = to_expr(e)
-            analyze('MatrixTable.group_rows_by', e, self._row_indices)
-            groups.append((k, e))
 
-        return GroupedMatrixTable(self, groups, self._row_indices)
+            if key in new_keys:
+                raise ExpressionException("method 'group_rows_by' found duplicate field: {}".format(key))
+            new_keys.append(key)
```

Comment:
I think this case is handled wrong -- suppose I pass ds.foo.bar.gene. what happens? Let's make sure that case is tested.